src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -233,6 +236,21 @@ constexpr std::string_view json_draft_4_metaschema = R"json(
 }
 )json";
 
+result<json::Document> parse_json(std::string_view v);
+
+ss::future<> check_references(sharded_store& store, canonical_schema schema) {
+    for (const auto& ref : schema.def().refs()) {
+        try {
+            // Proxy for checking that the subject-version exists
+            co_await store.is_subject_version_deleted(ref.sub, ref.version);
+
```

Comment:
Do we allow soft deleted subject versions as references? (Equivalent question: do we need to check that the return value of `store.is_subject_version_deleted(...) ==  false`?)

Reply:
> Do we allow soft deleted subject versions as references? (Equivalent question: do we need to check that the return value of `store.is_subject_version_deleted(...) == false`?)

Good point. According to the code you recently changed, no.

Reply:
Fixed